core/jvm/src/test/scala/zio/RTSSpec.scala

```diff
@@ -1209,6 +1212,20 @@ class RTSSpec(implicit ee: ExecutionEnv) extends TestRuntime {
       b      <- effect.await
     } yield b) must_=== 42
 
+  def testFirstSuccessOfValues =
+    unsafeRun(IO.firstSuccessOf(IO.fail(42), List(IO.succeed(24))).either) must_=== Right(24)
+
+  def testFirstSuccessOfFailures = {
+    implicit val d
+      : Diffable[Left[Int, Nothing]] = Diffable.eitherLeftDiffable[Int] //    TODO: Dotty has ambiguous implicits
+    unsafeRun(ZIO.firstSuccessOf(IO.fail(24).delay(10.millis), List(IO.fail(24))).either) must_=== Left(24)
+  }
+
+  def testFirstSuccessOfFailuresOneSuccess =
+    unsafeRun(ZIO.firstSuccessOf(IO.fail(42), List(IO.succeed(24).delay(1.millis))).either) must_=== Right(
+      24
```

Comment:
Due to jitter, it's best to use `IO.never` for the left value.

Reply:
Will change, thanks

Reply:
Ah, sorry, I misread the test. I think the delay is fine. 👍